src/projectviewer/projectviewermodel/recentprojectsmodel.cpp

```diff
@@ -0,0 +1,32 @@
+#include "recentprojectsmodel.h"
+
+#include <QDebug>
+#include <QIcon>
+
+RecentProjectsModel::RecentProjectsModel(const ProgramSettings &settings, QObject *parent)
+    :mSettings{settings}
```

Comment:
When inherits from QObject or related classes, you should inherit them correctly, not to broke Qt tree parent-child mechanism. Your constructor should look like:
`RecentProjectsModel::RecentProjectsModel(const ProgramSettings &settings, QObject *parent)
    : QAbstractListModel{parent}, mSettings{settings} { ...`